Assets/Scripts/Terrain/HeightMap/LargeHeightMap.cs

```diff
@@ -0,0 +1,109 @@
+using UnityEngine;
+using System;
+
+/// <summary>
+/// Large Height map of a specified size
+/// </summary>
+[RequireComponent(typeof(AbstractHeightMapGenerator))]
+public class LargeHeightMap : HeightMapComponent {
+
+    /// <summary>
+    /// Size of the map. Will be a square with edges of this length.
+    /// </summary>
+    public int mapSize = 1024;
+
+    /// <summary>
+    /// Minimum map height
+    /// </summary>
+    public int minHeight = 0;
+
+    /// <summary>
+    /// Maximum map height
+    /// </summary>
+    public int maxHeight = 256;
+
+    /// <summary>
+    /// Saved height map values.
+    /// </summary>
+    private float[] heightMap;
+        
+
+    /// <summary>
+    /// Initialized the heightMap values using the HeightMapGenerator
+    /// </summary>
+    public void GenerateHeightMap() {
+        AbstractHeightMapGenerator mapGen = GetComponent<AbstractHeightMapGenerator>();
+        this.heightMap = mapGen.CreateHeightMap(this.mapSize);
+        int heightMapRange = maxHeight - minHeight;
+
+        for (int x = 0; x < this.mapSize; x++) {
+            for (int y = 0; y < this.mapSize; y++) {
+                this.heightMap[x + y * this.mapSize] = heightMap[x + y * this.mapSize] * heightMapRange + minHeight;
```

Comment:
use getIndex(x,y) here

Reply:
Done, I forgot the function was private (not good to use public functions in constructor in java, been doing too much java)